bake/remote.go

```diff
@@ -23,13 +25,42 @@ type Input struct {
 }
 
 func ReadRemoteFiles(ctx context.Context, nodes []builder.Node, url string, names []string, pw progress.Writer) ([]File, *Input, error) {
-	var session []session.Attachable
+	var sessions []session.Attachable
 	var filename string
+
 	st, ok := dockerui.DetectGitContext(url, false)
 	if ok {
-		ssh, err := controllerapi.CreateSSH([]*controllerapi.SSH{{ID: "default"}})
-		if err == nil {
-			session = append(session, ssh)
+		var sshSpecs []*controllerapi.SSH
+		if v := os.Getenv("BUILDX_BAKE_GIT_SSH"); v != "" {
+			var err error
+			sshSpecs, err = buildflags.ParseSSHSpecs([]string{v})
+			if err != nil {
+				return nil, nil, errors.Wrapf(err, "failed to parse BUILDX_BAKE_GIT_SSH")
+			}
+		} else {
+			sshSpecs = []*controllerapi.SSH{{ID: "default"}}
+		}
+		if ssh, err := controllerapi.CreateSSH(sshSpecs); err == nil {
+			sessions = append(sessions, ssh)
+		}
+
+		var gitAuthSecrets []*controllerapi.Secret
+		if _, ok := os.LookupEnv("BUILDX_BAKE_GIT_AUTH_TOKEN"); ok {
```

Comment:
Probably out of scope, but unclear to me how the controller takes `Env` keys as input. Doesn't the controller have its own env if it is running in another process?

Reply:
Hum good point, it seems to work when connecting to local controller. I don't think there is another process except when using `--invoke` and setting envs through invoke config: https://github.com/docker/buildx/blob/a7d59ae3328bc5da57ba7f73a65a0b5f36ffe1bb/controller/pb/controller.proto#L199